packages/cubejs-api-gateway/src/jwk.ts

```diff
@@ -51,7 +56,7 @@ export type JWKsFetcherOptions = Pick<BackgroundMemoizeOptions<any, any>, 'onBac
 
 export const createJWKsFetcher = (jwtOptions: JWTOptions, options: JWKsFetcherOptions) => {
   const fetchJwkUrl = asyncMemoizeBackground(async (url: string) => {
-    const response = await asyncRetry(() => fetch(url), {
+    const response = await asyncRetry(async () => fetch(url, { agent: await getHttpAgentForProxySettings() }), {
```

Comment:
@KSDaemon It feels that we need to cache the result of `getHttpAgentForProxySettings.` Let's do a local variable cache in `@cubejs-backend/shared`. WDYT?

Reply:
Yeah, makes sense!